Improvised Carbines

shazbot194 opened the pull request:
This PR adds in 3 new improvised guns, the 9mm Improvised  Carbine, the .45 Improvised  Carbine, and the .357 Improvised  Carbine. These guns act just like the improvised shotgun, as they are single shot, but can only be constructed via the crafting menu. 

The Improvised 9mm Carbine
![bbihfxngsucvfppnog2vgq](https://user-images.githubusercontent.com/22019537/28156149-fc5e1f28-675d-11e7-8c5c-3cbc2d623e4a.png)
![lpzjkwpjqeqyis7jily-zg](https://user-images.githubusercontent.com/22019537/28156147-fc5ce176-675d-11e7-9548-fffbe3713093.png)
![u6fuuuoitu67lqyx0x_fow](https://user-images.githubusercontent.com/22019537/28156148-fc5d2e1a-675d-11e7-8ff4-ed91a4cdfa27.png)
The 9mm version is the weakest shot for shot, and easiest to make. It replaces the pipe with rods, and additionally needs 1 sheet of wood to make. 
If you add a plasteel to the completed gun, it will add a pull through magazine that can hold five rounds, but its a little cumbersome to use. (fire delay is 5) (pictures of magazine in a post bellow)
Currently, in this PR, there are two kinds of 9mm, default and rubber. Rubber 9mm does 5 brute and 30 stamina, and the default ammo does a flat 20 brute damage.
If you somehow find a universal suppressor, the 9mm carbine can accept it.

The Improvised .45 Carbine
![fzdfzedlrk_3bz7sxc60tq](https://user-images.githubusercontent.com/22019537/28156262-7f84c6cc-675e-11e7-8fdd-699e6e449dac.png)
![vr6qfbint-i34ihg1vgqvq](https://user-images.githubusercontent.com/22019537/28156264-7f88351e-675e-11e7-85de-b73b6ac0206a.png)
![c3r2g7j_qxit3o2paanowq](https://user-images.githubusercontent.com/22019537/28156263-7f8737f4-675e-11e7-9c72-5fdf42808722.png)
The .45 does more damage and does a little bit of stamina damage. It needs more rods and wood in addition to metal and a new part called the heatsink tube, found inside any hacked autolathe.
Much like the 9mm, a sheet of plasteel can be use to add a magazine to this gun, but it sadly only holds three bullets, and the locking mechanism is quite sticky. (fire delay is 10) (pictures of magazine in a post bellow)
Currently, in this PR, there are two kinds of 45, default and rubber. Rubber 45 does 5 brute and 65 stamina, and the default ammo does a 20 brute and 65 stamina damage.
Much like the 9mm carbine, it can also accept a suppressor.

The Improvised .357 Carbine
![z54zd4bqqbwjhaf4zcnnua](https://user-images.githubusercontent.com/22019537/28156219-512d9e98-675e-11e7-9aa6-b98324d42457.png)
![dqmj8n1vttwmb0ojlid1qq](https://user-images.githubusercontent.com/22019537/28156221-5131c626-675e-11e7-8bcb-53e40cdde099.png)
![l_hl2ojjrtwd0zips7zxnw](https://user-images.githubusercontent.com/22019537/28156220-512f8be0-675e-11e7-9a38-7864da8d6d8c.png)
This big beast of a gun can shoot the famous .357 rounds, but can still only load one round at a time, however it can be adjusted with a screwdriver to shoot .38 rounds. It also requires the heatsink tube and even more wood and rods then the .45 carbine and needs a sheet of plasteel and a sheet of silver to make.
Currently, .357 does 60 damage, and .38 does 5 brute and a short stun.
Unlike the other two carbines, it cannot accept a suppressor.

9mm and .45 rubber ammo boxes.
![cel6d-actighak-r-mlc-a](https://user-images.githubusercontent.com/22019537/28197200-c1745ee6-6801-11e7-9dc5-1623c8fc53d4.png)![9yetav3mqiorycyxf7ullw](https://user-images.githubusercontent.com/22019537/28197201-c1770d8a-6801-11e7-9d73-c252761fe86e.png)
Now you can make rubber 9mm and rubber 45 ammo boxes in the autolathe. I also changed the stamina damage of the .45 rubber bullets to do the same stamina damage as their lethal counterparts.


This PR also has a small change to improvised guns as they no longer spawn with ammo.

🆑 Shazbot
Adds: Adds in 3 new Improvised guns, the Improvised  9mm Carbine, the Improvised .45 Carbine, and the Improvised .357 Carbine.
Adds: Adds 9mm and .45 rubber bullets to the autolathe.
Fix: Both 9mm and .45 now use the proper ammo box.
Change: 45 rubber now does the same stamina damage as normal 45 bullets.
Change: Improvised guns no longer spawn with ammo.
/🆑

Fox-McCloud commented:
>giving crew access to makeshift 357 revolvers.

Yeah, no.

shazbot194 commented:
More like a single shot .357 pistol.

Birdtalon commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -359,6 +359,57 @@
+					to_chat(user, "<span class='warning'>You can't modify it!</span>")
+					return
+				magazine.caliber = "38"
+				desc = "Essentially a tube that aims .357 bullets, but shortened to shoot 38."
```
>.38

Also are .38 bullets a thing? Where do you get them from?

shazbot194 replied:
They are what the detective uses.

Birdtalon replied:
I learned something then today!

Birdtalon commented on code/modules/projectiles/ammunition/magazines.dm:
```diff
@@ -114,6 +114,27 @@
+	name = "pistol carbine internal magazine"
+	ammo_type = /obj/item/ammo_casing/c45
+	caliber = ".45"
+	max_ammo = 1
```
I'd like to see the improvised weapons have more than a 1 max ammo as it essentially makes them a real pain to use and are never a viable choice for antagonists. Perhaps you can craft a larger capacity magazine with other things?

shazbot194 replied:
I would be up for a craft-able magazine extentions, I think a cap of 5-8 rounds would be great for the 9mm.

Birdtalon replied:
Yeah maybe for the 9mm, perhaps not so much for the more powerful weapons.

shazbot194 replied:
I think 3-5 for the .45 wouldn't be too much either, but as a side though, we might want to add in the .45 rubber ammo to autolathes.

taukausanake replied:
Use plasteel for magazine extensions then. Maybe rods too but I can't think of anything extra aside from those

Birdtalon commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -359,6 +359,57 @@
+	desc = "Essentially a tube that aims .45 bullets."
+	icon_state = "i45"
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/improvised/i45
+	sawn_desc = "Placeholder."
```
Just don't let these three placeholders be forgotten.

shazbot194 replied:
This is honestly why its [WIP] because I don't want these to be forgotten.

The-Driker commented:
> giving crew access to makeshift 357 revolvers.

Yes, and double yes.

SamHPurp commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -359,6 +359,57 @@
+	desc = "Essentially a tube that aims 9mm bullets."
+	icon_state = "i9"
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/improvised/i9
+	sawn_desc = "Placeholder."
```
`Placeholder` - a reminder to fix before `[WIP]` to be removed.

SamHPurp commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -359,6 +359,57 @@
+	desc = "Essentially a tube that aims .357 bullets."
+	icon_state = "i357"
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/improvised/i357
+	sawn_desc = "Placeholder."
```
`Placeholder` - a reminder to fix before `[WIP]` to be removed.

taukausanake commented:
I think it's more along the lines of letting crew members (or per rules antags) arm themselves in more interesting manners. Underpowering compared to other tools they could use yeah but this is something needed all the same. That universial receiver is not very "universal" when it can only make one gun

imsxz commented:
I don't see the issue with "giving the crew .357"... .357 uses the same projectile that shotty slugs use, and the crew already has access to that via makeshift shotguns.

Rumiluntti commented:
Oh fuck no.
We are not giving 357's to crew

Spartan6 commented:
Is it me or is there no difference between the sprites?

taukausanake commented:
@Spartan6 There is not A LOT of difference between the sprites. I think that's an inherited issue with the receiver and wooden stock

Twinmold93 commented:
Like how people are saying don't give .357 to crew, yet they already have access as others point out (and makeable from the autolathe). The ability to make more makeshift guns is nice, gives people more of a chance to arm themselves when shit hits the fan, but also gives antags more possibilities. Obviously it'd be possession of a lethal weapon, so that should in theory help desaude people from just making them (you already rarely see the improvised shotgun or pneumatic canon being made [see NEVER]).

Regen1 commented:
I'd like for there to be more to it than just "More of each item needed to make X", because it doesn't really make it a challenge to make a higher tier.

Find some other items to change up the recipes to make them more distinct, that way people won't just always go for the best one (bar the plaststeel)

shazbot194 commented:
I would be more then open to suggestions on what to add into the recipes to make them harder to make, although with the introduction of magazine extensions, they all should be about on the same playing field power wise, at the cost of having similar components needed.

shazbot194 commented:
Ok, so I got the magazines working properly. They act just like normal revolver style magazines in that they retain the spent ammo until reloaded or emptied. They are both made by using a sheet of plasteel on the gun. The gun has to be empty or else it goes off in your face. To acount for larger ammo pools, both guns receive a slower rate of fire when converted to magazine fed. The 9mm now has a delay of 5, and the .45 has a delay of 10. The .357 still cannot be upgraded with a magazine at all.
![rvzets9kqxwyt-gma4krmw](https://user-images.githubusercontent.com/22019537/28192547-6f698de2-67e3-11e7-8d55-a2a42fc46b1b.png)
![jntljuynqe2xoqe8ouduia](https://user-images.githubusercontent.com/22019537/28192549-6f80d178-67e3-11e7-99f8-8f2e95496686.png)
![cfghdxait4sakayiifjydw](https://user-images.githubusercontent.com/22019537/28192548-6f7fee48-67e3-11e7-9698-a094068ab3c5.png)
![4zh92m5gt8ge0nyygxjhow](https://user-images.githubusercontent.com/22019537/28192551-738d5d18-67e3-11e7-96cc-cde5a7459135.png)
![1mgo5mottmeak4ye-lszaq](https://user-images.githubusercontent.com/22019537/28192552-738e2d06-67e3-11e7-8caf-e6c08dbe37bf.png)
![lhfza3rptisluljcfjv7aq](https://user-images.githubusercontent.com/22019537/28192553-7391f9e0-67e3-11e7-9a46-b720784ea983.png)

shazbot194 commented:
So I have added more materials to make the .45 and .357 carbines. They both need a new part called the heatsink tube, which a tube made to disperse heat, it can be made in any hacked autolathe. I also added one silver sheet to the materials of the .357 carbine to make it just a little harder to make.

SamHPurp commented on code/modules/crafting/guncrafting.dm:
```diff
@@ -15,6 +15,12 @@
 
+/obj/item/weaponcrafting/heatsink
+	name = "heatsinked tube"
+	desc = "A rather think tube with indents on it meant to disperse heat."
```
`think` -> `thin`

but its not written very well...

maybe:

`A remarkably thin tube marked with indents designed to disperse heat.` or something similar

shazbot194 replied:
That is a lot better, and I honestly don't know how that k got in there.

ktoma36 commented:
Perhaps make attachable things like wire cutter bayonets, duct-taped flashlights, jury rigged can suppressors, etc? I would love to go full ghetto commando plus it would give some more functionality to the guns (maybe more attachments for conventional weaponry would be cool later on as well :3)

shazbot194 commented:
If we ever do have a redo of the gun code that makes it a lot easier to do that with less snowflaking, I would 100% be up for having more makeshift attachments on these.

shazbot194 commented:
So that last addition adds in the 9mm rubber and .45 rubber to the autolathe. It also makes the .45 and .45 rubber do the same stamina damage.

taukausanake commented:
> wire cutter bayonets

Yeah, because those are so sharp when you poke someone with them. I'd rather it be a kitchen knife or a glass shank. Bonus points if you can have multiple sprites for them
> can suppressors

Or you can use potatoes apparently :^)

ktoma36 commented:
More ways to subdue people nonlethally, always a good thing

FlattestGuitar commented:
I'd rather see just one type of these carbines, frankly. They all look incredibly similar and are straight upgrades over each other. Not much reason to go with 9mm ammo if you can go for the .45.

Also, suppressors and wirecutter bayonets sound dope, but that will need the sprites to be redone.

shazbot194 commented:
Well, I could add in a system for suppressors and bayonets, it would be a little snowflake, probably to the horror of the maintainers. And I think with the addition of magazines, there is a valid reason why you would want the 9mm over the .45, and why you would want both of those over the .357. Although I could increase the mag size of the 9mm to make it more appealing. The rather funny thing is, 9mm and .45 do about the same damage to eachother, the .45 just has a slight edge in pure damage and does stamina damage.

ktoma36 commented:
People underestimate the power of 9mm, I usually rock only a pistol during nuke ops rounds just because of how much ammo you can scrounge up. Also 9mm is cheaper in the autolaths, making it a more savvy choice when resources are low (which will be if people really need to resort to makeshift carbines in a station defense scenario)

shazbot194 commented:
Due to popular request, I have added in a new ability for the 9mm and the .45.
![yqxxn78nrhseb7-iol9uea](https://user-images.githubusercontent.com/22019537/28400879-818e191e-6cc2-11e7-8aad-09cd3c8aefee.png)
You can now attach a universal suppressor to the 9mm and the .45. You cannot make these by yourself and are more meant for people who buy a suppressor or get lucky and find one.
This last pull adds in some code optimizations and now has less repeated code as well as improving the code already in place.
It also adds in the sawn off desc for the carbines. The .357 is in reference to a video of a guy on a lawnmower and a .44 magnum.

shazbot194 commented:
Ok, @Fox-McCloud, @tigercat2000, and @Crazylemon64, this pull is ready for review. I am more then willing to make changes if they are requested, but I think right now it is in a rather good spot, as each gun brings something to the table, the 9mm having higher ammo capacity with a magazine, the .45 doing some damage with a healthy dose of stamina damage, and the .357 doing the most damage, and all having less utility then the improvised shotgun, with the .357 doing the same damage as the improvised shotgun loaded with slugs.

jughu commented:
Basically det revolver and other fuckibg guns but in shotgun addition atleast make them nonnnn shotguns?

shazbot194 commented:
REEEEEEEEEEEEEEEE

Fox-McCloud commented:
Conflicts.

Rejected though; we really don't want more improvised guns at the moment.

ktoma36 commented:
;_;
